doozerlib/comment_on_pr.py

```diff
@@ -0,0 +1,79 @@
+from ghapi.all import GhApi
+from doozerlib.pushd import Dir
+from dockerfile_parse import DockerfileParser
+
+
+class CommentOnPr:
+    def __init__(self, distgit_dir: str, token: str):
+        self.distgit_dir = distgit_dir
+        self.token = token
+        self.owner = None
+        self.repo = None
+        self.commit = None
+        self.gh_client = None  # GhApi client
+        self.pr_url = None
+        self.pr_no = None
+
+    def list_comments(self):
+        """
+        List the comments in a PR
+        """
+        # https://docs.github.com/rest/reference/issues#list-issue-comments
+        return self.gh_client.issues.list_comments(issue_number=self.pr_no, per_page=100)
+
+    def check_if_comment_exist(self, comment):
+        """
+        Check if the same comment already exists in the PR
+        """
+        issue_comments = self.list_comments()
+        for issue_comment in issue_comments:
+            if issue_comment["body"] == comment:
+                return True
+        return False
+
+    def post_comment(self, comment):
+        """
+        Post the comment in the PR if the comment doesn't exist already
+        """
+        # https://docs.github.com/rest/reference/issues#create-an-issue-comment
+        if not self.check_if_comment_exist(comment):
```

Comment:
`self.check_if_comment_exist` calls `self.list_comments` internally, which requires `self.pr_no` is set. However `self.pr_no` is not set until `self.get_pr_from_commit` is called.

Reply:
Can't believe I missed that! Apologies! Thank you for pointing it out!